shared/actions/chat/attachment.js

```diff
@@ -25,7 +25,10 @@ function * onShareAttachment ({payload: {message}}: Constants.ShareAttachment):
 function * onSaveAttachmentNative ({payload: {message}}: Constants.SaveAttachment): SagaGenerator<any, any> {
   const {filename, messageID, conversationIDKey} = message
   if (filename && messageID) {
-    const path = downloadFilePath(filename)
+    let path = tmpFile(filename)
+    if (filename.match(/[^/]+\.(jpg|png|gif|jpeg|bmp)$/) == null) {
```

Comment:
instead of having this check in 4 different places we should make a helper in constants/chat or something